recipes/libnghttp2/all/conanfile.py

```diff
@@ -32,34 +37,34 @@ class Nghttp2Conan(ConanFile):
         "with_asio": False,
     }
 
-    generators = "cmake", "cmake_find_package", "pkg_config"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
         if not (self.options.with_app or self.options.with_hpack or self.options.with_asio):
-            del self.settings.compiler.cppstd
-            del self.settings.compiler.libcxx
+            try:
+                del self.settings.compiler.cppstd
+            except Exception:
+                pass
+            try:
+                del self.settings.compiler.libcxx
+            except Exception:
+                pass
         if not self.options.with_app:
             del self.options.with_jemalloc
 
+    def layout(self):
+        cmake_layout(self)
```

Comment:
```suggestion
        cmake_layout(self, src_folder="src")
```

Reply:
Could you explain, why is it needed here? Nothing in recipe relies on the source folder name. Majority of recipes uses default `.` directory too. Checked [the docs](https://docs.conan.io/en/latest/reference/conanfile/tools/layout.html#conan-tools-layout-predefined-layouts), nothing sheds light.

Reply:
Unfortunately, I don't see any docs in CCI for this particular practice, although I think there's good reason for it. All the package templates under `docs` do this. It's not set in the `test_package`'s, though. @prince-chrismc Care to chime in? Maybe we should add a quick note about why to do this in the docs?

Reply:
To avoid mixing source files with cmake cache. When creating a recipe, it will copy all source files to <package_id>/build and will run cmake configure on the same folder, mixing everything. When installing a package it does not matter, but for who is developing or reviewing a package, it's a bit cleaner to read all files. So, using `src` Conan will use that folder to copy only source files and will not run cmake there.

Reply:
Thank you for explanations. It's more useful for local development. At least it's not needed to pass source_folder everywhere to cmake. If it is a preferred way for CCI, I can add a hook for checking.

Reply:
> Thank you for explanations. It's more useful for local development. At least it's not needed to pass source_folder everywhere to cmake. If it is a preferred way for CCI, I can add a hook for checking.

Pylint please!!! All the code checks are going there now :) 


If you you python ast please put lots of comments so we can learn 🙏